### PR TITLE
:bug: Analytic View results refresh fix

### PR DESCRIPTION
**Pull request by aldebaran30701**
### Description of the Change
This change tracks the last current graph loaded with the view, and checks before reloading the view if a complete view reset is necessary.

### Alternate Designs
N/A

### Why Should This Be In Core?
Bug fix

### Benefits
Keeps the results pane open when using the same graph and switching views

### Possible Drawbacks
N/A

### Verification Process
Test opening with a graph, test opening without a graph, test switching of graph when view is minimized, test results pane consistency with same graphs.

### Applicable Issues
#718

### CODE REVIEW

File: CoreAnalyticView/src/au/gov/asd/tac/constellation/views/analyticview/AnalyticViewTopComponent.java [hunk 1]
```diff
@@ -74,6 +75,7 @@
     private final AnalyticViewPane analyticViewPane;
     private final AnalyticController analyticController;
     private boolean suppressed = false;
+    private Graph currentGraph = null;
```
Comment (arcturus2) [inline CoreAnalyticView/src/au/gov/asd/tac/constellation/views/analyticview/AnalyticViewTopComponent.java:78]: There is potential for a memory leak to appear if the `currentGraph` is not set to `null` when the graph closes. I am speculating but I will do some testing to confirm @aldebaran30701.

Comment (aldebaran30701) [reply CoreAnalyticView/src/au/gov/asd/tac/constellation/views/analyticview/AnalyticViewTopComponent.java:78]: How does it sound if I store the current graph's ID instead?

Comment (arcturus2) [reply CoreAnalyticView/src/au/gov/asd/tac/constellation/views/analyticview/AnalyticViewTopComponent.java:78]: That sounds like a better idea though we haven't depended on the graph ID for much so I am not sure if it is the best approach. If it works then that does sound like a really good way forward and perhaps we can adopt it across more of the framework.

I remember seeing the graph ID change after plugins run (but I need to double check my memory). If that's the case (which I don't think so) would be a problem.

Comment (aldebaran30701) [reply CoreAnalyticView/src/au/gov/asd/tac/constellation/views/analyticview/AnalyticViewTopComponent.java:78]: Hi @cygnus-x-1,  Would you be able to confirm for us if the graph id changes between plugin runs? Or subsequently offer a different way of tracking the current graph without memory leaks.

Comment (antares1470) [reply CoreAnalyticView/src/au/gov/asd/tac/constellation/views/analyticview/AnalyticViewTopComponent.java:78]: I don't know whether or not it helps avoid memory leaks but having played with the rest api previously, I believe the graph Id remains the same between plugin runs (even between Constellation loads)

Comment (arcturus2) [reply CoreAnalyticView/src/au/gov/asd/tac/constellation/views/analyticview/AnalyticViewTopComponent.java:78]: >...(even between Constellation loads)

Not sure on this bit @antares1470. The graph UUID is not stored in the star file so I believe a random one is issued on load.

Comment (cygnus-x-1) [reply CoreAnalyticView/src/au/gov/asd/tac/constellation/views/analyticview/AnalyticViewTopComponent.java:78]: I believe a graph id should be reliable as long as that graph is in memory.

Comment (antares1470) [reply CoreAnalyticView/src/au/gov/asd/tac/constellation/views/analyticview/AnalyticViewTopComponent.java:78]: Right. Like I said, it's been a while so I was only going off the top of my head

Comment (aldebaran30701) [reply CoreAnalyticView/src/au/gov/asd/tac/constellation/views/analyticview/AnalyticViewTopComponent.java:78]: > If the graph is closed and opened again, it'll have a new id

I'd think that would cross over into a future state implementation. This (keeping the id) would work for the sake of not clearing the results table.

**Pull request merged**